r2-core/src/main/java/com/linkedin/r2/transport/http/client/ratelimiter/CallbackWrapper.java

```diff
@@ -0,0 +1,9 @@
+package com.linkedin.r2.transport.http.client.ratelimiter;
+
+import com.linkedin.common.callback.Callback;
+import com.linkedin.common.util.None;
+
+
+public interface CallbackWrapper {
```

Comment:
Glad you asked! Before this change, we'd pass in a ConstantQpsRateLimiter that we could extend with whatever additional functionality we desired through an inheritance programming model. The move to supply a ConstantQpsRateLimiterFactory changes this to a composition model, where we're now instead passing the functionality we need in the form of a wrapper.

Name dropping class names from LI internal libraries: We'll no longer be able to use ICAwareCallbackCircularBuffer in the package that pulls in rest.li, and will instead be passing a ICAwareCallbackWrapper.

Reply:
1. Instead of passing in a callback wrapper, have you considered passing a FactoryFactory so that you can use your own factory to generate ratelimiter you want?
2. Please do not define new interface. Use CallbackAdaptor instead.

Reply:
+1 for using com.linkedin.common.callback.CallbackAdapter.

Also as Chao mentioned, using FactoryFactory pattern has the advantage that you will be able to define an `ICAwareRateLimiterFactory` in LI internal libraries to generate IC aware rate limiter

Reply:
Thanks for pushing back on this. The latest diff uses a Supplier instead of a static factory, which I think takes care of the concerns raised here. Let me know what you think.